sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5443,7 +5443,16 @@ pi_result piextEventCreateWithNativeHandle(pi_native_handle NativeHandle,
   // Assume native event is host-visible, or otherwise we'd
   // need to create a host-visible proxy for it.
   (*Event)->HostVisibleEvent = *Event;
-
+  // A regular pi_event created by piEventCreate() starts with its reference
+  // count = 2 because of our explicit call to piEventCreate() at the end of
+  // createEventAndAssociateQueue().
+  // Then, this ref count becomes zero when the event is completed and RT
+  // calls piEventRelease().
+  // However, an interop event created in this function would go through
+  // the same life cycle, meaning it will be released twice.
```

Comment:
I realize the interop PI event created here would be released by the SYCL RT (at SYCL event's destructor) but what will wait until the event is completed?

Reply:
We don't know. I think we should not make any assumption about interop event.

Reply:
So we will not do the wait in SYCL RT, and hence the extra retain will cause this PI event to never be released.

Reply:
In fact, I just checked SYCL_PI_TRACE and found that interop event is different than regular event. 
Regular event created in piEventCreate() has ref count=2 at the beginning and then decremented twice before it is deallocated.
However, interop event is not created by createEventAndAssociateQueue(), so its ref count is 1, and piEventRelease() is called by RT.
So, we don't have memory leak for interop event.